### PR TITLE
aacgain: 1.9.0 -> 2.0.0

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Supersedes #191114 and #191144.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/aacgain/default.nix [hunk 1]
```diff
@@ -1,68 +1,36 @@
...
-    make
-    runHook postBuild
-  '';
-
-  installPhase = ''
-    install -D aacgain/aacgain "$out/bin/aacgain"
-  '';
+  nativeBuildInputs = [
+    cmake
+    autoconf
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/aacgain/default.nix:24]: autoconf and cmake? Thats normally not done

File: pkgs/applications/audio/aacgain/default.nix [hunk 2]
```diff
@@ -1,68 +1,36 @@
...
-    repo = "aacgain";
-    rev = "7c29dccd878ade1301710959aeebe87a8f0828f5";
-    sha256 = "07hl432vsscqg01b6wr99qmsj4gbx0i02x4k565432y6zpfmaxm0";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "dgilman";
+    repo = pname;
+    rev = version;
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/aacgain/default.nix:18]: We should have those libraries in nixpkgs https://github.com/dgilman/aacgain/blob/master/.gitmodules

Comment (wegank) [reply pkgs/applications/audio/aacgain/default.nix:18]: I'm afraid using shared libraries in nixpkgs isn't trivial, as we have something like

```
#include "src/impl.h" //in mp4v2
```

everywhere in the source code. This also prevents dropping autotools.

Comment (SuperSandro2000) [reply pkgs/applications/audio/aacgain/default.nix:18]: Than I would suggest to at least symlink our versions to receive patches and updates.

Comment (wegank) [reply pkgs/applications/audio/aacgain/default.nix]: Done.

**Pull request merged**